src/Aspire.Dashboard/Utils/CertificateUtil.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Cryptography.X509Certificates;
+
+namespace Aspire.Dashboard;
+
+internal static class CertificateUtil
+{
+    public static X509CertificateCollection GetFileCertificate(string filePath, string? password, ILogger logger)
+    {
+        var certBytes = File.ReadAllBytes(filePath);
+
+        var certContentType = X509Certificate2.GetCertContentType(certBytes);
+
+        if (certContentType is X509ContentType.Pkcs12)
+        {
+            return [X509CertificateLoader.LoadPkcs12(certBytes, password)];
+        }
+        else
+        {
+            if (password is not null)
+            {
+                logger.LogDebug("Resource service certificate {FilePath} has type {Type} which does not support passwords, yet a password was configured. The certificate password will be ignored.", filePath, certContentType);
+            }
+
+            return [X509CertificateLoader.LoadCertificate(certBytes)];
```

Comment:
_Technically_ speaking, you have a functionality regression here.  The old code would load any of

* A single certificate
  * Which is 99% useless, but SslStream has a feature where it looks in the cert stores to see if maybe you happened to have specified a certificate that you already know about and have a private key for... (which might only be on Windows?)
* A PKCS#7 SignedCms file.
  * Where it loads the certificate used by the first signer, or fails if it can't find one (or there is no first signer).
  * And then is the same as the "plain certificate" case.
* A Windows SerializedCert blob (Windows only)
  * This "might" have an associated private key, just by virtue of it might have written down "and my private key is named {whatever}" AND that key happens to be available on this machine/user-context.
  * Otherwise it's just a "plain certificate"
  * But basically no one uses this format.
* An Authenticode-signed file (exe, dll, msu, cab, etc). (Windows-only)
  * It extracts the certificate of the first signer.
  * Sounds a bit like SignedCms, eh? Oh, wait, Authenticode uses SignedCms.  Less coincidence, more "same code".
* A PKCS#12/PFX file
  * _Probably_ has a certificate in it with a private key, but, might not.

The new code will throw if it encounters any of those three middle formats, because the new loader is "one method (group), one file format".  The middle ones don't _really_ make sense in context for you, so they're not really important... but this is, technically, marginally, a breaking change.

https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview7/libraries.md#changes-to-x509-certificate-loading shows an equivalent to what `new X509Certificate2(bytesOrPath, password, flags)` does.